### PR TITLE
Add config parameter for read-only SQLite databases

### PR DESCRIPTION
**Pull request by Ma27**
See https://github.com/doctrine/dbal/pull/4292

### CODE REVIEW

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -201,6 +201,7 @@ private function configureDbalDriverNode(ArrayNodeDefinition $node) : void
                 ->scalarNode('application_name')->end()
                 ->scalarNode('charset')->end()
                 ->scalarNode('path')->end()
+                ->booleanNode('read_only')->end()
```
Comment (ostrolucky) [inline DependencyInjection/Configuration.php:204]: should define info explaining this, similarly as is done for `persistent` option

**Review by ostrolucky — CHANGES_REQUESTED** [review]
Requires documentation, xsd updates

**Comment by Ma27** [comment]
@ostrolucky thanks, fixed :)

The build failure appears unrelated as it's also broken on `master`.

**Comment by Ma27** [comment]
Just pushed an update: the DBAL maintainers preferred to not have this kind of logic in their library however I think that it's still useful to have this setting accessible and exposed in a config layer like this bundle. For further references to the original discussion, please take a look at the commit message :)

**Comment by ostrolucky** [comment]
Sorry, I agree with @morozov. This adds just yet another abstraction level to real driver option that doesn't work for any other drivers. Rather confusing if users can just use existing constant for a driver option for same thing. Also, we generally do not add driver options here that do not exist in DBAL, I would recommend to wait until feature is accepted in DBAL first before working on its integration here. Thanks for proposing though!

**Pull request closed without merging**